examples/flux1.py

```diff
@@ -0,0 +1,962 @@
+# This file incorporates code from the following:
+# Github Name                    | License | Link
+# black-forest-labs/flux         | Apache  | https://github.com/black-forest-labs/flux/tree/main/model_licenses
+
+import os
+import re
+import time
+from dataclasses import dataclass
+from glob import iglob
+from PIL import Image
+import argparse
+
+import math
+from typing import Callable
+
+from tinygrad import Tensor, nn, dtypes, TinyJit
+from tinygrad.nn.state import safe_load, load_state_dict
+from tinygrad.helpers import fetch, tqdm
+
+from extra.models.clip import FrozenClosedClipEmbedder
+from extra.models.t5 import T5Embedder
+
+def TensorIdentity(x: Tensor) -> Tensor:
+  return x
+
+# https://github.com/black-forest-labs/flux/blob/main/src/flux/math.py
+
+def attention(q: Tensor, k: Tensor, v: Tensor, pe: Tensor) -> Tensor:
+  q, k = apply_rope(q, k, pe)
+
+  x = Tensor.scaled_dot_product_attention(q, k, v)
+  x = x.rearrange("B H L D -> B L (H D)")
+
+  return x
+
+def rope(pos: Tensor, dim: int, theta: int) -> Tensor:
+  assert dim % 2 == 0
+  scale = Tensor.arange(0, dim, 2, dtype=dtypes.float64, device=pos.device) / dim
```

Comment:
So for this to work on OSX you need to do this. Would be nice to platform gate this for OSX hardware


  scale = Tensor.arange(0, dim, 2, dtype=dtypes.float32, device=pos.device) / dim

Reply:
Thanks for spotting. Reference implementation just uses float64, diffusers implementation downcasts only for metal (as you suggest). Will take a look at how much of a difference it makes (might cause problems for long prompts?)

Reply:
it doesn't make much - fp32 looks better sometimes tbh